pkg/frontend/querymiddleware/split_by_instant_interval_test.go

```diff
@@ -23,279 +23,290 @@ import (
 )
 
 func TestQuerySplittingCorrectness(t *testing.T) {
-	var (
-		numSeries          = 1000
-		numStaleSeries     = 100
-		numHistograms      = 1000
-		numStaleHistograms = 100
-		histogramBuckets   = []float64{1.0, 2.0, 4.0, 10.0, 100.0, math.Inf(1)}
-	)
+	for _, startString := range []string{
+		"2020-01-01T03:00:00.100Z",
+		"2020-01-01T03:00:00Z",
```

Comment:
Hum, couldn't we maybe move this to a different test? 🤔 And keep this one with the `time.Now()`?
If not or it would be harder to test all the cases, I would suggest to also add the `time.Now()` to this map.

Reply:
Why would testing at `time.Now()` be better? It's non-reproducible.
I can add that here, but unit tests shouldn't depend on their environment IMO.

Reply:
Not better, just adding more diversity to the tests as it is also done in the sharding middleware: https://github.com/grafana/mimir/blob/main/pkg/frontend/querymiddleware/querysharding_test.go
My suggestion was to also add the `time.Now()` in addition to these time dates, but I'm ok with not adding them 🙂